assets/home/main.js

```diff
@@ -61,4 +69,23 @@ const Main = () => {
 	);
 };
 
+/**
+ * Filter to add the notices section based on the EditorNotices component.
+ *
+ * @param {JSX.Element} previous The previous element to be added
+ * @return {JSX.Element} The new top of the Sensei Home page, with the editor notices as a column.
+ */
+function addNotices( previous ) {
+	return (
+		<>
+			{ previous }
+			<Col as="section" className="sensei-home__section" cols={ 12 }>
```

Comment:
Do you think this `Col` should really be added here? Or maybe it should be part of the `Main` component? My question is because I think other components shouldn't worry about columns while adding custom things here, or should it be?

Reply:
Hey there. Just now I saw your answer.. :facepalm: sorry for the delay!. 

I think this is a tough thing to answer, to be honest. We'll probably have more than one filter being added here, like the ones being added in Sensei Pro, and so I feel like having control over the columns being used is...good? 

Also, if I move this `<Col>`  to the place where we call `applyFilters`, we will have more than one component being added in the same `<Col>`, which is... a little weird?  Imagine the licensing block and the notices on the same block on the same column, for instance..

Personally, what worries me the most is the idea of having to nest each filter under fragments to make it "extendable". Like, if we add 3 filters like this one:

```js
let count = 1;
function addFilter(previous) {
    return ( <> { previous } Filter {count++} </>);
}
addFilter( 'sensei.home.top', 'sensei/home/top/add-filter-1', addFilter );
addFilter( 'sensei.home.top', 'sensei/home/top/add-filter-2', addFilter );
addFilter( 'sensei.home.top', 'sensei/home/top/add-filter-3', addFilter );
```

We would have something like this in the React structure:

```jsx
<>
   Filter 3
   <>
      Filter 2
      <> 
           Filter 1
       </>
   </>
</>
```

Which...doesn't sound very beautiful? But at the same time, I have no idea on how to improve this. :disappointed:

Reply:
Jumping in to ask: why don't we consider the notices a separate thing from the Sensei Home top hook?

I mean, we can add the Sensei Home hook (for whatever content is added like the Licensing from Sensei Pro) and also have the EditorNotices always included. I don't see the point of adding `EditorNotices` by using the hook.

Reply:
Hey @aaronfc. My initial ideas was to be able to position the notices appropriately. Like, if we want, with the current code we can show things both above or below the `EditorNotices` component, which is something that we MAY need to do considering Sensei Pro.

In the same way, we can just replace the EditorNotices block altogether, by not adding `previous` to the component output. This means that we can replace it totally with another thing, like the activation component, for instance.

I'm not sure, though. 

Does it make any sense? :smile:

Reply:
> Which...doesn't sound very beautiful? But at the same time, I have no idea on how to improve this. 😞

Have you considered the [`SlotFill`](https://developer.wordpress.org/block-editor/reference-guides/components/slot-fill/)? Probably it fits better in this case instead of a filter?

> we can show things both above or below the EditorNotices component, which is something that we MAY need to do considering Sensei Pro

Hm! That's a good discussion. Maybe we could have 2 things (filter or slot): `"AfterTitle"` and `"AfterNotices"`. Another option could be using PHP hooks here, keeping the notices and the page title out of this React tree. So we have more freedom to extend it later (with PHP or React - Creating a new React tree). I remember we talked about using PHP hooks here, but I don't remember what was our conclusion. 🤔

Another thing I was checking related to this. If we continue with a filter, I think the default value for the filter could be `null` instead of an empty Fragment. WDYT?

Reply:
Hey @renatho, thanks for the reply. 

> Have you considered the [SlotFill](https://developer.wordpress.org/block-editor/reference-guides/components/slot-fill/)? Probably it fits better in this case instead of a filter?

That's the first thing I looked into, actually. :joy: 

I abandoned the idea as soon as I thought: "okay, where should I put the `<SlotFillProvider />`, and, even more, how will the other components inject things here?". 

Maybe we could use a `SlotFill` + filters to do this? I'm not sure.

> Hm! That's a good discussion. Maybe we could have 2 things (filter or slot): "AfterTitle" and "AfterNotices". Another option could be using PHP hooks here, keeping the notices and the page title out of this React tree. So we have more freedom to extend it later (with PHP or React - Creating a new React tree). I remember we talked about using PHP hooks here, but I don't remember what was our conclusion. :thinking:

If we put the notices on the PHP side, I think we may struggle a little bit with the Notices API/styling, so I'm not sure if that's worth.

About using PHP hooks, I think we were referring to the activation/renewal sections from Sensei Pro, which will be injected by PHP (on the Sensei Pro side) conditionally depending on whether we need to show those sections or not, and using PHP so we can avoid the idea of always inserting stuff via JS and then checking via JS. At least, what's what I remember.

> Another thing I was checking related to this. If we continue with a filter, I think the default value for the filter could be null instead of an empty Fragment. WDYT?

Good catch, fixed in 4d41ebe33695a68df1727483822a2dc7357e1e6a.

Reply:
Thank you! =)

About the `Col`, I still think that other parts adding the filters wouldn't need to add the `Col` by themselves, mainly because it will probably always be the `<Col cols={ 12 }>`. But I think that's fine and we can figure out later, when using it what's the best. :)